ocaml/xapi/xapi_host.ml

```diff
@@ -1346,9 +1346,23 @@ let apply_edition_internal  ~__context ~host ~edition ~additional =
     | Missing_connection_details -> raise Api_errors.(Server_error(missing_connection_details, []))
     | License_checkout_error s -> raise Api_errors.(Server_error(license_checkout_error, [s]))
   in
-  Db.Host.set_edition ~__context ~self:host ~value:edition';
-  let features = Features.of_assoc_list features' in
-  copy_license_to_db ~__context ~host ~features ~additional
+
+  let update_feature fname fenabled host =
+    Db.Feature.get_all_records ~__context |>
+      List.filter (fun (_, r) -> r.API.feature_name_label = fname && r.API.feature_host = host) |>
+    function
+    | [] -> Db.Feature.create ~__context ~uuid:(Uuid.to_string (Uuid.make_uuid ()))
+              ~ref:(Ref.make ()) ~name_label:fname ~name_description:""
+              ~enabled:fenabled ~experimental:true ~version:"1.0"
+              ~host
+    | [(rf, _)] -> Db.Feature.set_enabled ~__context ~self:rf ~value:fenabled
+    | _ -> raise Api_errors.(Server_error(multiple_feature_instance, [fname; Ref.string_of host]))
+  in
+  List.iter (fun (x, y) -> update_feature x y host) new_ed.experimental_features;
```

Comment:
You don't need the `host` parameter to `update_feature`, because it is in scope. Also, if you define the function as `let update_feature (feature, enabled)`, then you can just do `List.iter update_feature new_ed.experimental_features`.